common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -386,6 +444,46 @@ protected final boolean runAllTasksFrom(Queue<Runnable> taskQueue) {
         }
     }
 
+    /**
+     * What ever tasks are present in {@code taskQueue} when this method is invoked will be {@link Runnable#run()}.
+     * @param taskQueue the task queue to drain.
+     * @return {@code true} if at least {@link Runnable#run()} was called.
+     */
+    private boolean runExistingTasksFrom(Queue<Runnable> taskQueue) {
+        return taskQueue.offer(BOOKEND_TASK) ? runExistingTasksUntilBookend(taskQueue)
+                                             : runExistingTasksUntilMaxTasks(taskQueue);
+    }
+
+    private boolean runExistingTasksUntilBookend(Queue<Runnable> taskQueue) {
+        Runnable task = pollTaskFrom(taskQueue);
+        if (task == BOOKEND_TASK) {
+            return false;
+        }
+        for (;;) {
+            safeExecute(task);
+            task = pollTaskFrom(taskQueue);
+            if (task == BOOKEND_TASK) {
```

Comment:
I am correct to say that we should never have `task == null` here and above ?

Reply:
I wouldn't expect this because we don't allow `null` to be offered to the queue, and we only enter this method if we successfully added `BOOKEND_TASK` to the queue. Am I missing something?

Reply:
sounds about right... Just wanted to ensure I read it correctly. Maybe an assert would clarify ?

Reply:
>  Maybe an assert would clarify ?

We can if you prefer, however if `null` does come up we will NPE and log an error.

Reply:
Let us just add a comment... can you do it ?

Reply:
done